pkg/api_load/api_manager.go

```diff
@@ -0,0 +1,230 @@
+
+// add apiLoader by ApiLoadType
+func (al *ApiManager) AddApiLoader(config model.ApiConfig) {
+	if config.File != nil {
+		al.ApiLoadTypeMap[File] = NewFileApiLoader(WithFilePath(config.File.FileApiConfPath))
+	}
+	if config.Nacos != nil {
+		al.ApiLoadTypeMap[Nacos] = NewNacosApiLoader(WithNacosAddress(config.Nacos.Address))
+	}
+}
+
+// nolint
+func (al *ApiManager) GetApiLoad(apiLoadType ApiLoadType) (ApiLoader, error) {
+	if apiLoader, ok := al.ApiLoadTypeMap[apiLoadType]; ok {
+		return apiLoader, nil
+	}
+	return nil, errors.New(fmt.Sprintf("can't load apiLoader for :%s", apiLoadType))
+}
+
+// start to load apis using apiLoaders stored in ApiLoadTypeMap
+func (al *ApiManager) StartLoadApi() error {
+	for _, loader := range al.ApiLoadTypeMap {
+		err := loader.InitLoad()
+		if err != nil {
+			logger.Warn("proxy init api error:%v", err)
+			break
+		}
+	}
+
+	if al.limiter == nil {
+		return errors.New("proxy won't hot load api since limiter is null.")
+	}
+
+	for _, loader := range al.ApiLoadTypeMap {
+		changeNotifier, err := loader.HotLoad()
+		if err != nil {
+			logger.Warn("proxy hot load api error:%v", err)
+			break
+		}
+
+		go func() {
+			for {
+				select {
+				case _, ok := <-changeNotifier:
+					if !ok {
+						logger.Debug("changeNotifier of apiloader was closed!")
+						return
+					}
+					al.AddMergeTask()
+					break
+				}
+			}
+		}()
+	}
+	return nil
+}
+
+// store a message to mergeTask to notify calling DoMergeApiTask
+func (al *ApiManager) AddMergeTask() error {
+	select {
+	case al.mergeTask <- struct{}{}:
+		logger.Debug("added a merge task, waiting to merge api.")
+		break
+	case <-time.After(5 * time.Second):
+		logger.Errorf("add merge task fail:wait timeout.")
+		break
+	}
+	return nil
+}
+
+// to merge apis to store in ads.Notice that limiter will limit frequency of merging.
+func (al *ApiManager) SelectMergeApiTask() (err error) {
+	for {
+		select {
+		case <-al.limiter.C:
+			if len(al.mergeTask) > 0 {
+				_, err = al.DoMergeApiTask()
+				if err != nil {
+					logger.Warnf("error merge api task:%v", err)
+				}
+			}
+			//al.limiter.Reset(time.Second)
+			break
+		default:
+			time.Sleep(al.rateLimiterTime / 10)
+			break
+		}
+	}
+	return
+}
+
+// merge apis
+func (al *ApiManager) DoMergeApiTask() (skip bool, err error) {
+	al.mergeLock.Lock()
+	defer al.mergeLock.Unlock()
+	wait := time.After(time.Millisecond * 50)
+	select {
+	case <-wait:
+		logger.Debug("merge api task is too frequent.")
+		skip = true
+		return
+	case <-al.mergeTask:
+		// If apiLoadType is File,then try covering it's apis using other's apis from registry center
+		multiApisMerged := make(map[string]model.Api, 8)
+		var sortedApiLoader []int
+		sortedApiLoaderMap := make(map[int]ApiLoadType, len(al.ApiLoadTypeMap))
+		for apiLoadType, loader := range al.ApiLoadTypeMap {
+			sortedApiLoader = append(sortedApiLoader, loader.GetPrior())
+			sortedApiLoaderMap[loader.GetPrior()] = apiLoadType
+		}
+
+		sort.Ints(sortedApiLoader)
```

Comment:
instead of this sorting, will it be better to cerate a sortableApiLoaders type and  implement Len, Swap, Less interface to do the sorting?